File: app/src/main/java/com/bojie/weatherBo/ui/DailyForecastActivity.java
package com.bojie.weatherbo.ui;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.v7.app.ActionBarActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import com.bojie.weatherbo.R;
import com.bojie.weatherbo.adapter.DayAdapter;
import com.bojie.weatherbo.util.UnitConvert;
import com.bojie.weatherbo.weather.Day;

import java.text.DecimalFormat;
import java.util.Arrays;

import butterknife.ButterKnife;
import butterknife.InjectView;

public class DailyForecastActivity extends ActionBarActivity {

    private Day[] mDays;

    @InjectView(android.R.id.list)
    ListView mListView;
    @InjectView(android.R.id.empty)
    TextView mEmptyTextView;
    @InjectView(R.id.tv_LocationLabel)
    TextView mLocationLabel;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_daily_forecast);
        ButterKnife.inject(this);

        Intent intent = getIntent();
        Parcelable[] parcelables = intent.getParcelableArrayExtra(MainActivity.DAILY_FORECAST);
        mDays = Arrays.copyOf(parcelables, parcelables.length, Day[].class);

        DayAdapter adapter = new DayAdapter(this, mDays);
        mListView.setAdapter(adapter);
        mListView.setEmptyView(mEmptyTextView);
        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String dayOfTheWeek = mDays[position].getDayOfTheWeek();
                String conditions = mDays[position].getSummary();

                double TempInF = mDays[position].getTemperatureMax();
                double TempInC = UnitConvert.fahrenheitToCelsius(TempInF);
                DecimalFormat df = new DecimalFormat();
                df.setMaximumFractionDigits(2);
                String highTemp = df.format(TempInF) + "";
                if (MainActivity.mButtonUnitConvert.getText() == "F") {
                    highTemp = df.format(TempInF) + "";
                } else if(MainActivity.mButtonUnitConvert.getText() == "C"){
                    highTemp = df.format(TempInC) + "";
                }
                String message = String.format("On %s the high will be %s and it will be %s",
                        dayOfTheWeek,
                        highTemp,
                        conditions);

                Toast.makeText(getApplicationContext(), message, Toast.LENGTH_LONG).show();

            }
        });
        String cityName = intent.getStringExtra(MainActivity.CITY_NAME);
        mLocationLabel.setText(cityName);

    }
}

File: app/src/main/java/com/bojie/weatherBo/ui/DailyForecastActivity.kt
package com.bojie.weatherbo.ui

import android.content.Intent
import android.os.Bundle
import android.os.Parcelable
import android.view.View
import android.widget.AdapterView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bojie.weatherbo.R
import com.bojie.weatherbo.adapter.DayAdapter
import com.bojie.weatherbo.databinding.ActivityDailyForecastBinding
import com.bojie.weatherbo.util.UnitConvert
import com.bojie.weatherbo.weather.Day
import java.text.DecimalFormat
import java.util.Arrays

class DailyForecastActivity : AppCompatActivity() {

    private lateinit var mDays: Array<Day>
    private lateinit var binding: ActivityDailyForecastBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDailyForecastBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        val intent = intent
        val parcelables = intent.getParcelableArrayExtra(MainActivity.DAILY_FORECAST)
        mDays = Arrays.copyOf(parcelables, parcelables!!.size, Array<Day>::class.java)
        
        val adapter = DayAdapter(this, mDays)
        binding.list.adapter = adapter
        binding.list.emptyView = binding.empty
        binding.list.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            val dayOfTheWeek = mDays[position].dayOfTheWeek
            val conditions = mDays[position].summary
            
            val tempInF = mDays[position].temperatureMax
            val tempInC = UnitConvert.fahrenheitToCelsius(tempInF)
            val df = DecimalFormat().apply {
                maximumFractionDigits = 2
            }
            
            val highTemp = when (MainActivity.mButtonUnitConvert.text) {
                "F" -> df.format(tempInF)
                "C" -> df.format(tempInC)
                else -> df.format(tempInF)
            }
            
            val message = String.format(
                "On %s the high will be %s and it will be %s",
                dayOfTheWeek,
                highTemp,
                conditions
            )
            
            Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
        }
        
        val cityName = intent.getStringExtra(MainActivity.CITY_NAME)
        binding.tvLocationLabel.text = cityName
    }
} 